Index: Connection Test/Connection Test/app/src/main/java/com/example/myapplication/Connectivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Connection Test/Connection Test/app/src/main/java/com/example/myapplication/Connectivity.java	(revision 7b4c69cb68ab8db806a1d252cb7d0f9a5de1a94a)
+++ Connection Test/Connection Test/app/src/main/java/com/example/myapplication/Connectivity.java	(date 1582791591041)
@@ -127,6 +127,6 @@
         rv.setAdapter(adapter);
         // Set layout manager to position the items
         rv.setLayoutManager(new LinearLayoutManager(this));
-    }
 
-}
+    }
+}
\ No newline at end of file
Index: Connection Test/Connection Test/app/src/main/res/layout/recycler_items.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Connection Test/Connection Test/app/src/main/res/layout/recycler_items.xml	(revision 7b4c69cb68ab8db806a1d252cb7d0f9a5de1a94a)
+++ Connection Test/Connection Test/app/src/main/res/layout/recycler_items.xml	(date 1582791595267)
@@ -22,4 +22,5 @@
         android:paddingRight="16dp"
         android:textSize="10sp"
         />
+
 </LinearLayout>
\ No newline at end of file
Index: Connection Test/Connection Test/app/src/main/java/com/example/myapplication/ChatActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Connection Test/Connection Test/app/src/main/java/com/example/myapplication/ChatActivity.java	(revision 7b4c69cb68ab8db806a1d252cb7d0f9a5de1a94a)
+++ Connection Test/Connection Test/app/src/main/java/com/example/myapplication/ChatActivity.java	(date 1582948592057)
@@ -10,6 +10,11 @@
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 import com.google.firebase.database.DataSnapshot;
@@ -21,6 +26,7 @@
 public class ChatActivity extends AppCompatActivity {
     Button b;
     private DatabaseReference databaseRef;
+    GoogleSignInClient mGoogleSignInClient;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -66,8 +72,29 @@
     }
 
     public  void logout(View view){
+// Configure Google Sign In
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestEmail()
+                .build();
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        mGoogleSignInClient = GoogleSignIn.getClient(this, gso);
 
         new SessionManagement(ChatActivity.this).remove();
+        mGoogleSignInClient.signOut()
+                .addOnCompleteListener(this, new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                        if(task.isSuccessful()) {
+                            Log.d("DB", "Logged out Gaccnt");
+                        }else{
+                            Log.d("DB", "No Gaccount");
+                        }
+                    }
+                });
+
+
         Intent intent = new Intent(ChatActivity.this, MainActivity.class);
         startActivity(intent);
         finish();
Index: Connection Test/Connection Test/app/src/main/java/com/example/myapplication/ContactsAdapter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Connection Test/Connection Test/app/src/main/java/com/example/myapplication/ContactsAdapter.java	(revision 7b4c69cb68ab8db806a1d252cb7d0f9a5de1a94a)
+++ Connection Test/Connection Test/app/src/main/java/com/example/myapplication/ContactsAdapter.java	(date 1582805317161)
@@ -65,7 +65,7 @@
         TextView textView = viewHolder.nameTextView;
         textView.setText(connect.username);
         Button button = viewHolder.messageButton;
-        button.setText("B");
+        button.setText("Connect");
     }
 
     // Returns the total count of items in the list
Index: Connection Test/Connection Test/gradle/wrapper/gradle-wrapper.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>windows-1252
===================================================================
--- Connection Test/Connection Test/gradle/wrapper/gradle-wrapper.properties	(revision 7b4c69cb68ab8db806a1d252cb7d0f9a5de1a94a)
+++ Connection Test/Connection Test/gradle/wrapper/gradle-wrapper.properties	(date 1582854971771)
@@ -1,6 +1,6 @@
-#Mon Jan 20 19:27:29 IST 2020
+#Fri Feb 28 07:26:11 IST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
Index: Connection Test/Connection Test/.idea/gradle.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Connection Test/Connection Test/.idea/gradle.xml	(revision 7b4c69cb68ab8db806a1d252cb7d0f9a5de1a94a)
+++ Connection Test/Connection Test/.idea/gradle.xml	(date 1582944528024)
@@ -1,15 +1,19 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="GradleMigrationSettings" migrationVersion="1" />
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
-        <compositeConfiguration>
-          <compositeBuild compositeDefinitionSource="SCRIPT" />
-        </compositeConfiguration>
+        <option name="testRunner" value="PLATFORM" />
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
+        <option name="modules">
+          <set>
+            <option value="$PROJECT_DIR$" />
+            <option value="$PROJECT_DIR$/app" />
+          </set>
+        </option>
         <option name="resolveModulePerSourceSet" value="false" />
-        <option name="testRunner" value="PLATFORM" />
       </GradleProjectSettings>
     </option>
   </component>
Index: Connection Test/Connection Test/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Connection Test/Connection Test/build.gradle	(revision 7b4c69cb68ab8db806a1d252cb7d0f9a5de1a94a)
+++ Connection Test/Connection Test/build.gradle	(date 1582854971731)
@@ -7,7 +7,7 @@
         
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.3'
+        classpath 'com.android.tools.build:gradle:3.6.0'
         classpath 'com.google.gms:google-services:4.3.2'  // Google Services plugin
         
         // NOTE: Do not place your application dependencies here; they belong
